Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -299,6 +306,12 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
 
         try self.testSession.approveAskToBuyTransaction(identifier: transaction.identifier)
 
+        // In JWS mode, transaction takes a bit longer to be processed after `approveAskToBuyTransaction`
+        // We need to wait so `restorePurchases` actually posts it.
+        if self.usesStoreKit2JWS {
+            try? await Task.sleep(nanoseconds: DispatchTimeInterval.milliseconds(500).nanoseconds)
```

Comment:
🥶🙈 This is a recipe for more flakiness. 
Can we use something like `waitUntilUnfinishedTransactions`? (or a similar method)